connectome_interpreter/external_map.py

```diff
@@ -160,25 +159,191 @@ def map_to_experiment(df, dataset=None, custom_experiment=None):
+                -'cbar_len': float, default=0.75
+        dpi : int, default=72
+            Dots per inch for the output figure. Standard is 72 for screen/SVG/PDF.
+            Use higher values (e.g., 300) for print-quality output.
+        custom_colorscale : list, default=None
+            Custom colorscale for the heatmap. If None, defaults to white-to-blue
+            colorscale [[0, "rgb(255, 255, 255)"], [1, "rgb(0, 20, 200)"]].
+
+    Returns:
+        fig : go.Figure
+    """
+    # Default styling and sizing parameters to use if not specified.
+    default_style = {
+        "font_type": "arial",
+        "markerlinecolor": "rgba(0,0,0,0)",  # transparent
+        "linecolor": "black",
+        "papercolor": "rgba(255,255,255,255)",
+    }
+
+    default_sizing = {
+        "fig_width": 260,  # units = mm
+        "fig_height": 220,  # units = mm
+        "fig_margin": 0,
+        "fsize_ticks_pt": 20,
+        "fsize_title_pt": 20,
+        "markersize": 18,
+        "ticklen": 15,
+        "tickwidth": 5,
+        "axislinewidth": 3,
+        "markerlinewidth": 0.9,
+        "cbar_thickness": 20,
+        "cbar_len": 0.75,
+    }
+
+    # If style is provided, update default_style with user values
+    if style is not None:
+        default_style.update(style)
+    style = default_style
+
+    if sizing is not None:
+        default_sizing.update(sizing)
+    sizing = default_sizing
+
+    # Constants for unit conversion
+    POINTS_PER_INCH = 72  # Typography standard: 1 point = 1/72 inch
+    MM_PER_INCH = 25.4    # Standard conversion: 1 inch = 25.4 mm
+    
+    # sizing of the figure and font
+    pixelsperinch = dpi  # Use the provided DPI value
+    pixelspermm = pixelsperinch / MM_PER_INCH
+    area_width = (sizing["fig_width"] - sizing["fig_margin"]) * pixelspermm
+    area_height = (sizing["fig_height"] - sizing["fig_margin"]) * pixelspermm
+    fsize_ticks_px = sizing["fsize_ticks_pt"] * (1 / POINTS_PER_INCH) * pixelsperinch
+    fsize_title_px = sizing["fsize_title_pt"] * (1 / POINTS_PER_INCH) * pixelsperinch
+
+    # Convert index values (formatted as '-12,34') into separate x and y
+    # coordinates
+    coords = [tuple(map(int, idx.split(","))) for idx in df.index]
+    x_vals, y_vals = zip(*coords)  # Separate into x and y lists
+
+    # initiate plot
+    fig = go.Figure()
+    fig.update_layout(
+        autosize=False,
+        height=area_height,
+        width=area_width,
+        margin={"l": 0, "r": 0, "b": 0, "t": 0, "pad": 0},
+        paper_bgcolor=style["papercolor"],
+        plot_bgcolor=style["papercolor"],
+    )
+    fig.update_xaxes(
+        showgrid=False, showticklabels=False, showline=False, visible=False
+    )
+    fig.update_yaxes(
+        showgrid=False, showticklabels=False, showline=False, visible=False
+    )
+    # Symbol number to choose to plot hexagons
+    symbol_number = 15
+
+    # Get the coordinates of all columns in the medulla:
```

Comment:
Hi @leburnett, could I check if col_coords (Nern2024) is actually needed here, because x and y values can come from x_vals, y_vals, which came from the 'x' and 'y' columns of ME-columnar-cells-hex-location.csv? Many thanks, Yijie